phoenix-core/src/main/java/org/apache/phoenix/hbase/index/Indexer.java

```diff
@@ -800,5 +804,62 @@ public static void enableIndexing(HTableDescriptor desc, Class<? extends IndexBu
     properties.put(Indexer.INDEX_BUILDER_CONF_KEY, builder.getName());
     desc.addCoprocessor(Indexer.class.getName(), null, priority, properties);
   }
+
+    /**
+     * stop indexer when region split, but split failed, and then rollback, open region.
+     * this need init builder and writer,recoveryWriter.
+     *
+     * @param env
+     */
+  private void initBuilderAndWriter(RegionCoprocessorEnvironment env) {
+      // init index build manager
+      try {
+          this.builder = new IndexBuildManager(env);
+      } catch (IOException e) {
+          LOG.warn("failed to new index build manager", e);
```

Comment:
Catching the exception here, and letting the function continue doesn't seem right, as we'll run into problems later.
Why not just let them bubble up, and handle them at the caller ?

Reply:
> Catching the exception here, and letting the function continue doesn't seem right, as we'll run into problems later.
> Why not just let them bubble up, and handle them at the caller ?

the postOpen() of hbase coprocessor can not bubble up this catch, only print log in this.